year18/src/day09.rs

```diff
@@ -1,8 +1,42 @@
 // Potential improvements:
 //
+use std::collections::VecDeque;
+use regex::Regex;
 
-pub fn day09(_input_lines: &[Vec<String>]) -> (String, String) {
-    let answer1 = 0;
+
+pub fn day09(input_lines: &[Vec<String>]) -> (String, String) {
+    let re = Regex::new(r"(\d+) players; last marble is worth (\d+) points").unwrap();
+    let caps = re.captures(&input_lines[0][0]).unwrap();
+    let player_count = caps.get(1).unwrap().as_str().parse::<i32>().unwrap();
+    let last_marble_value = caps.get(2).unwrap().as_str().parse::<u64>().unwrap() * 100;
+
+    let mut circle  = VecDeque::new();
+    let mut player_queue = VecDeque::new();
+    for _ in 0..player_count {
+        player_queue.push_back(0);
+    }
+    circle.push_back(0);
+    circle.push_back(2);
+    circle.push_back(1);
+    for marble_value in 3..=last_marble_value {
+        let mut player = player_queue.pop_front().unwrap();
+
+        if marble_value % 23 == 0 {
+            player += marble_value;
+            circle.rotate_right(8);
+            player += circle.pop_back().unwrap();
+            circle.rotate_left(2);
+
+        } else {
+            circle.push_back(marble_value);
+            circle.rotate_left(1);
+        }
+        // println!("{:?}", circle);
+        
+        player_queue.push_back(player);
+    }
+
+    let answer1 = player_queue.iter().max().unwrap();
     let answer2 = 0;
```

Comment:
Note, I just used the same code from part 1 for part 2 so didn't bother making things fit nicely with the framework.

Reply:
I'd suggest it's good practice to actually solve both parts at the same time rather than just hacking the code to update it (modulo time constraints).  It should be fairly easy to use a sub-function here to run both parts.